homeassistant/components/mqtt/models.py

```diff
@@ -91,17 +93,23 @@ def _convert_outgoing_payload(
 
             return payload
 
-        if self._attr_command_template is None:
+        if self._command_template is None:
             return value
 
-        values = {"value": value}
+        values: dict[str, Any] = {"value": value}
         if self._entity:
             values[ATTR_ENTITY_ID] = self._entity.entity_id
             values[ATTR_NAME] = self._entity.name
+            values.setdefault(
```

Comment:
This won't work, there's no guarantee the caller is not passing in a new dict each time this is called.
Instead, do something like:
```py
if not self._template_state:
    self._template_state = template.TemplateStateFromEntityId(self._command_template.hass, self._entity.entity_id)
values[ATTR_THIS] = self._template_state
```

Reply:
Done